cli/azd/pkg/httputil/util.go

```diff
@@ -25,3 +27,12 @@ func ReadRawResponse[T any](response *http.Response) (*T, error) {
 
 	return instance, nil
 }
+
+// Handles and errors executing the http request
+func HandleRequestError(response *http.Response, err error) error {
+	if response == nil {
+		return fmt.Errorf("failed executing request: %w", err)
+	}
+
+	return runtime.NewResponseError(response)
+}
```

Comment:
What if `err == nil`?

Reply:
This `HandleRequestError(...)` should only be called in an error scenario when we have an error.  That is the case for all current call paths.

Reply:
Nvm, it's used from the caller